### DISCUSSION TITLE
v0.12 Dev Channel

### DISCUSSION THREAD

**Discussion by Blinue** [Blinue/Magpie#926] [Dev Channel]
v0.12 聚焦于：

1. 实现窗口化缩放 #135
2. 易用性改进

**Comment by sao321** [comment]
等待了N年的窗口化终于要来了，感谢大佬！

**Reply by Blinue** [reply]
大概要等很久，这个功能很难做，我不想做成 LS 那样简陋的窗口化

**Reply by RosaElzer** [reply]
win11 24h2
无法在这个系统使用

**Comment by Blinue** [comment]
#848 没了。@cunnyplapper 已注销

**Reply by hooke007** [reply]
这槽点太多了，以前只知道issue会消失，没想到pr都能没

**Reply by Blinue** [reply]
注销机制有点迷，变成 @ghost 是最好的。顺带一提 #721 又回来了，作者变了（改了账号名？为什么有一段时间是注销状态？）

**Reply by hooke007** [reply]
cunny 在这的提交之后重做了几次，这是5月的最终版magpie着色器
[CuNNy-67b027f.zip](https://github.com/user-attachments/files/15822255/CuNNy-67b027f.zip)

**Reply by Blinue** [reply]
新版本是同时缩放 RGB 通道，使用 https://www.ozone3d.net/public/jegx/201412/chroma-subsampling-test-4k-tv.png 对比差别比较明显。问题是文字边缘容易产生伪影，很难说比旧版本更好。

**Reply by hooke007** [reply]
https://github.com/funnyplanter/CuNNy
似乎换号回来了

**Reply by Blinue** [reply]
这么恶趣味的名字想换号也可以理解😅

**Reply by Blinue** [reply]
> https://github.com/funnyplanter/CuNNy

这个仓库的版本很不错，完全解决了伪影问题

**Reply by Blinue** [reply]
#848 和 #721 都没了

**Reply by hooke007** [reply]
这个倒是再次提醒我了，要把重要的帖子重新写在自己的主题名下，防止原帖因为OP的任何账户问题而消失

**Comment by shitake99** [comment]
CuNNy 有更新的版本请问可以加入吗？
https://github.com/funnyplanter/CuNNy/tree/master/magpie
![Screenshot 2025-02-20 194959](https://github.com/user-attachments/assets/2081bf16-ec74-4a6b-ae31-ea34d5e580ab)

**Reply by Blinue** [reply]
变体和之前不一样，不好升级。我现在对添加效果很谨慎，计划以后支持订阅仓库，将效果更新和主仓库解绑。

**Reply by Blinue** [reply]
已经集成新版本命名为 CuNNy2，因为它的效果确实很好，我想把它集成进 v0.12。订阅仓库至少在 v0.12 不会实现。

**Comment by lucksufe** [comment]
我想对整个桌面或者说显示区域而不是某个窗口进行后处理，可以借鉴这个项目或者修改某些部分就可以完成吗？

**Reply by Blinue** [reply]
可以。Graphics Capture 和 Desktop Duplication 都支持捕获屏幕，你需要修改捕获的部分和缩放窗口位置。

**Comment by Blinue** [comment]
Github Actions 编译 onnxruntime 在 patch 中失败了，排查了好久发现 Strawberry Perl 污染了 PATH，它包含一个旧版本或者自己编译的野生版本 patch.exe。https://github.com/actions/runner-images/issues/5459 还有一个很好玩的 meme🤣喜欢 patch 的人有福了

**Comment by hooke007** [comment]
我忘了magpie是支持还是不支持，读取effects目录里自行添加的着色器？

**Reply by Blinue** [reply]
要重新启动 magpie

**Reply by hooke007** [reply]
![image](https://github.com/user-attachments/assets/5b0f3ed9-f41d-49df-897a-87e14523162a)
https://github.com/Blinue/Magpie/issues/1125#issuecomment-2819929643 dpid 一直读取不出来我以为是没这个功能，直到试了试其它的

**Reply by hooke007** [reply]
```
	版本: v0.11.2
	管理员: 否
2025-04-23 19:26:34.711|error|EffectCompiler.cpp:1572|Compile|解析 Header 块失败
2025-04-23 19:26:34.735|info|AppSettings.cpp:231|Initialize|便携模式: 否
2025-04-23 19:29:25.627|info|XamlApp.cpp:271|_QuitWithoutMainWindow|程序退出
2025-04-23 19:29:30.860|info|main.cpp:77|wWinMain|程序启动
```

**Reply by hooke007** [reply]
`#include "StubDefs.hlsli"` 去掉就识别了

**Reply by Blinue** [reply]
最新开发版才支持，去掉这行 v0.11 就能用了。我把文件更新了

**Comment by Blinue** [comment]
https://github.com/microsoft/onnxruntime/pull/24456
onnxruntime 刚加了 TensorRT RTX 后端，据说能大幅减小体积而且优化更好。只支持 RTX 系列应该不是问题。支持 TensorRT 的消费级显卡只有 RTX 系列。

**Reply by hooke007** [reply]
我迅速询问了大佬的相关意见，似乎目前不太行 https://github.com/AmusementClub/vs-mlrt/issues/132#issuecomment-2829585586

**Reply by Blinue** [reply]
从代码看支持构建引擎，应该和传统 lean 版本不同。可惜用的 trt 版本还没发布，编译不了

**Reply by Blinue** [reply]
[TensorRT for RTX 已经发布](https://developer.nvidia.com/blog/nvidia-tensorrt-for-rtx-introduces-an-optimized-inference-ai-library-on-windows/)，SDK 下个月可用。总结一下：

1. 体积约 200MB（AOT 和 JIT 各 100MB），比 TensoRT 小得多。
2. 分为两个阶段：AOT 阶段生成中间引擎，完全在 CPU 上执行，用时约 15s；JIT 阶段在运行时动态优化。
3. 可以不指定形状，JIT 会自动优化。

性能没和 TensorRT 对比，但既然有 JIT 应该不会太差，但还是比不上最优配置的 TensorRT 就是了。

**Reply by Blinue** [reply]
微软发布了 [WinML](https://learn.microsoft.com/en-us/windows/ai/new-windows-ml/overview)，可以根据 GPU 自动选择后端，其中包括 TensorRT for RTX。目前部署太麻烦了，如果未来能集成进 OS 应该很有用。

**Reply by Blinue** [reply]
TensorRT for RTX 的 SDK 发布了，dll 体积只有 150MB

**Comment by rampaa** [comment]
I've developed an [overlay app](https://github.com/rampaa/JL) that marks itself as `Magpie.ToolWindow` in accordance with the current [documentation](https://github.com/Blinue/Magpie/blob/dev/docs/Interact%20with%20Magpie%20programally.md). Will that be unnecessary once v0.12 comes out?

**Reply by Blinue** [reply]
Yes, `Magpie.ToolWindow` is now obsolete, as scaling can continue in the background. The documentation will be updated after #1071 is merged.

**Comment by Blinue** [comment]
我有意让全屏和窗口缩放使用不同的配置文件，但 UI 不好设计

**Reply by hooke007** [reply]
<img width="500" src="https://github.com/user-attachments/assets/66bec04f-2495-4b58-939e-28c6fafc9bdc" />
加个按钮一键切换到窗口配置如何。全屏和窗口仍然共用左侧一个标签，这样避免不必要的配置条目臃肿

**Reply by Blinue** [reply]
问题在于全屏和窗口的配置很多时候要一起改，比如换个缩放模式要改两个地方就有点麻烦
